### PR TITLE
Catalog: Feature/preserve data

### PR DESCRIPTION
**Pull request by mateo951**
- Added preserve data logic
- Added fix for linters

### CODE REVIEW

File: lib/books/book_collection.rb [hunk 1]
```diff
@@ -1,10 +1,7 @@
 require_relative 'book'
 require_relative './../items/label'
-require_relative './../helpers/variables_interface'
 
 module BookCollection
-  include VariablesInterface
-
   def list_all_books
     puts "Your library don't have books yet. Add one first!" if @books.empty?
```
Comment (Felipeg005) [inline lib/books/book_collection.rb:7]: Great Syntaxis here 👍

File: lib/helpers/preserve_data.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def save_genres
+    base_folder = './lib/helpers/json'
+    genres_path = './lib/helpers/json/genres.json'
+
+    FileUtils.mkdir_p(base_folder) unless Dir.exist?(base_folder)
+    json_genres = @genres.map { |genre| genre.to_json }
+    File.write(genres_path, JSON.generate(json_genres))
+  end
```
Comment (Felipeg005) [inline lib/helpers/preserve_data.rb:33]: This code (line21 and 30) it's checking for json folder in all 'save' methods, maybe we can use it just once?

**Review by Felipeg005 — APPROVED** [review]
Hello @nelsinofrancisco and @mateo951  !

Great Job with this implementation!!!

## Approved✔️ 

Time to merge this branch. 😎 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] Consider the comment below.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._***

File: lib/helpers/preserve_data.rb (hunk 2 above)
Comment (mateo951) [reply lib/helpers/preserve_data.rb:33]: done! thanks

**Pull request merged**